CryptoPrices/Sources/Styleguide/Sources/Styleguide/SwiftUIHelpers/Preview.swift

```diff
@@ -0,0 +1,34 @@
+//  Preview.swift
+//  Home
+//
+//  Created by Doan Thieu on 04/12/2022.
+//
+
+// swiftlint:disable type_contents_order
+
+import SwiftUI
+
+public struct Preview<Content: View>: View {
+
+    let content: Content
+
+    public init(@ViewBuilder _ content: () -> Content) {
+        self.content = content()
+    }
+
+    public var body: some View {
+        Group {
+            content
+                .environment(\.colorScheme, .light)
+                .preferredColorScheme(.light)
+                .previewDevice("iPhone 14 Pro")
+                .previewDisplayName("Pro - Light mode")
+
+            content
+                .environment(\.colorScheme, .dark)
+                .preferredColorScheme(.dark)
+                .previewDevice("iPhone 14 Pro")
+                .previewDisplayName("Pro - Dark mode")
+        }
+    }
```

Comment:
Instead of disable `swiftlint` here with `swiftlint:disable type_contents_order`, I think we can just simply reorder the function here a bit as follow:
```suggestion
// swiftlint:disable type_contents_order

import SwiftUI

public struct Preview<Content: View>: View {

    let content: Content

    public var body: some View {
        Group {
            content
                .environment(\.colorScheme, .light)
                .preferredColorScheme(.light)
                .previewDevice("iPhone 14 Pro")
                .previewDisplayName("Pro - Light mode")

            content
                .environment(\.colorScheme, .dark)
                .preferredColorScheme(.dark)
                .previewDevice("iPhone 14 Pro")
                .previewDisplayName("Pro - Dark mode")
        }
    }
    
      public init(@ViewBuilder _ content: () -> Content) {
        self.content = content()
    }
```

Reply:
Thanks, @minhnimble 🙏 

I think everyone (including me at first) just simply reorder a little bit to satisfy `swiftlint` here. But the more I think about it, the more it annoys me 😅

That's because the SwiftUI `View` is special and different from other types, like data structures or view models, ... The SwiftUI `View` merely is a description of a view, and almost its code lives in the `body` computed property. And this `body` computed property is quite long depending on how complex the layout is. It isn't simply with one or a few lines of codes like `var httpMethod: String { "GET" }` as we normally see. As a result, the initializer, which is the very first API that we use, the one that tells us how we initialize, construct, and create a new instance easily gets lost at the very bottom of the view definition.

I did some research to conditionally opt out of this `type_contents_order` rule specifically for SwiftUI views but got no luck. This issue has been even raised on the `swiftlint` repo [here](https://github.com/realm/SwiftLint/issues/3548) but no progress so far.

But I think for now I can go with the reordering approach and let all of us experience this matter and have a further discussion along the way 🙏

Reply:
@Thieurom the `body` var is indeed a special and potentially long variable in SwiftUI for sure. At the same time, it is all coming down to the perspective of the developers and in this case, it can be seen that we have a slightly different view at the moment as I am not really having an issue with placing the init function a bit below 😅. For me, in our previous ViewModels that have huge logic with many states, we would need to create a lot of variables before the `init` call, right? At the same time, grouping variables and functions correspondingly allow much easy searching and organization, especially when we combine it with the `MARK` usage. Then we can use Xcode's feature to find what we want and later on where to add new vars/functions with ease 🙏:

![Screenshot 2022-12-06 at 10 08 47](https://user-images.githubusercontent.com/70877098/205802409-9fece737-bf8f-4dd8-8145-5cfe508c2f21.png)

-> Anyhow, if you really want to commit to placing the init function a bit earlier at the top level and mix them in between, setting `type_contents_order` rule on the file is still not a great approach because it is going to be a recurring problem. We will have to add this rule on every file that needs the `init` function, which might also introduce more potential issues that some new devs might forget/ be unaware of the requirement here. A better way could be disabling the rule directly on the `.swiftlint.yml` file for having a global effect 💪

Reply:
@minhnimble Maybe I'm a little bit unclear here 🙏  When I say `gets lost` it's not about how hard to navigate to (by means of `MARK` like you said, which I'm aware of and definitely agree to apply). What I just mean is the way I look at the code, it's more about aesthetics, easy in the eye I guess 😅 . And as I [provided](https://github.com/realm/SwiftLint/issues/3548), it's not only me who has the same experience, and the same feeling but there're some other guys too. So I think it's maybe just a matter of preference, which is subjective, so feel free to ignore it 😄
> A better way could be disabling the rule directly on the .swiftlint.yml file for having a global effect.

This rule should be encouraged to apply across the app, that's why I was trying to conditionally opt out specifically for SwiftUI views only, not globally but `swiftlint` does not support it now. Until then we don't any other way than to manually disable file by file. One quick way I can think of right now is using a code generator (like `sourcery` to prepend that disabling rule at the top of every SwiftUI file). But maybe I overthink it, and I'm totally fine with the reordering approach to avoid making things complicated.